x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetectorFactory.java

```diff
@@ -111,6 +114,12 @@ private void getCardinalitiesForFieldsWithConstraints(String[] index, DataFrameA
 
         SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().size(0).query(config.getSource().getParsedQuery());
         for (FieldCardinalityConstraint constraint : fieldCardinalityConstraints) {
+            for (FieldCapabilities fieldCaps : fieldCapabilitiesResponse.getField(constraint.getField()).values()) {
```

Comment:
🤔 

What if this is a multi-field?

If the field mapping is
```
"foo": {
   "type": "text",
   "keyword": {
      "type": "keyword"
   }
}
```
Do we prefer to use `foo.keyword` in this instance? I know that analytics chooses a preferred field-mapping but I am not sure that occurs before this call.

Reply:
In your example, we'll prefer `foo.keyword` indeed.
But this matters only for fields that have constraints.

At the moment, the only field with cardinality constraint is the `dependent_variable` which we'll take in whatever form the user types it in.

If we ever introduce other fields with cardinality constraints, we'll have to ensure they're aggregatable anyway.

Reply:
AH! We don't dynamically choose the best multi-field for the `dependent_variable`. In that case, :shipit: